ivy_tests/test_ivy/helpers/assertions.py

```diff
@@ -135,9 +138,23 @@ def value_test(
         )
     )
     # value tests, iterating through each array in the flattened returns
-    if not rtol:
+    if specific_tolerance_dict is not None:
+        for ret_np, ret_np_from_gt in zip(ret_np_flat, ret_np_from_gt_flat):
+            dtype = str(ret_np_from_gt.dtype)
+            if specific_tolerance_dict.get(dtype) is not None:
+                rtol = specific_tolerance_dict.get(dtype)
+            else:
+                rtol = TOLERANCE_DICT.get(dtype, 1e-03)
```

Comment:
You can change this to `rtol = TOLERANCE_DICT.get(dtype, 1e-03) if rtol is None else rtol` since I would expect `rtol` of all the other dtypes to be the value I provided in the `rtol` argument and since thats in higher priority

Reply:
@sherry30 I already added this logic when the rtol is `None` because if we pass in the tolerance dict we won't need the tolerance that is passed in by the user. It can be inferred from the global `TOLERANCE_DICT`

Reply:
Yes, but what I'm saying is the value passed by the user in the `rtol` argument is of higher priority than `TOLERANCE_DICT`. So we should infer from that instead if the specific dtype does not exist in `specific_tolerance_dict` and considering `rtol` was provided by the user.